tests/s25Main/UI/testIngameWindow.cpp

```diff
@@ -36,3 +39,67 @@ BOOST_AUTO_TEST_CASE(IngameWnd)
     wnd.SetMinimized(false);
     BOOST_TEST_REQUIRE(wnd.GetSize() == oldSize);
 }
+
+BOOST_AUTO_TEST_CASE(IwMapGenerator)
+{
+    enum
+    {
+        CTRL_BTN_BACK = 0,
+        CTRL_BTN_APPLY,
+        CTRL_TXT_LANDSCAPE,
+        CTRL_TXT_GOAL,
+        CTRL_TXT_IRON,
+        CTRL_TXT_COAL,
+        CTRL_TXT_GRANITE,
+        CTRL_TXT_RIVERS,
+        CTRL_TXT_MOUNTAIN_DIST,
+        CTRL_TXT_TREES,
+        CTRL_TXT_STONE_PILES,
+        CTRL_TXT_ISLANDS,
+        CTRL_PLAYER_NUMBER,
+        CTRL_MAP_STYLE,
+        CTRL_MAP_SIZE,
+        CTRL_MAP_TYPE,
+        CTRL_RATIO_GOLD,
+        CTRL_RATIO_IRON,
+        CTRL_RATIO_COAL,
+        CTRL_RATIO_GRANITE,
+        CTRL_RIVERS,
+        CTRL_MOUNTAIN_DIST,
+        CTRL_TREES,
+        CTRL_STONE_PILES,
+        CTRL_ISLANDS
+    };
+
+    uiHelper::initGUITests();
+    rttr::mapGenerator::MapSettings settings;
+    iwMapGenerator wnd(settings);
+    wnd.GetCtrl<ctrlComboBox>(CTRL_PLAYER_NUMBER)->SetSelection(2); // 4 players
+    wnd.GetCtrl<ctrlComboBox>(CTRL_MAP_TYPE)->SetSelection(2); // LandscapeDesc(2)
+    wnd.GetCtrl<ctrlComboBox>(CTRL_MAP_STYLE)->SetSelection(1); // MapStyle::Land
+    wnd.GetCtrl<ctrlComboBox>(CTRL_MAP_SIZE)->SetSelection(4); // 1024x1024
+    wnd.GetCtrl<ctrlComboBox>(CTRL_MOUNTAIN_DIST)->SetSelection(3); // MountainDistance::VeryFar
+    wnd.GetCtrl<ctrlComboBox>(CTRL_ISLANDS)->SetSelection(2); // IslandAmount::Many
+    wnd.GetCtrl<ctrlProgress>(CTRL_RATIO_GOLD)->SetPosition(25);
```

Comment:
Can you use different values? Bonus points for random ones, but likely overkill here. Reason is that the current test won't catch confusion of sliders